tests/test-parser.bsh

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env bash
+
+. "$(dirname "${BASH_SOURCE[0]}")/testlib.bsh"
+. "$(dirname "${BASH_SOURCE[0]}")/test_utils.bsh"
+
+VSI_COMMON_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")/.."; pwd)"
+. "${VSI_COMMON_DIR}/linux/parser.bsh"
+
+begin_test "Test parse url"
+(
+  setup_test
+
+  parse_url "https://hub.server.com"
+  [ "${just_url_parsed_scheme}" = 'https' ]
+  [ "${just_url_parsed_authority}" = 'hub.server.com' ]
+  [ "${just_url_parsed_host}" = 'hub.server.com' ]
+  [ "${just_url_parsed_port}" = '' ]
+  [ "${just_url_parsed_rpath}" = '' ]
+  [ "${just_url_parsed_query}" = '' ]
+  [ "${just_url_parsed_fragment}" = '' ]
+
+  parse_url "https://hub.server.com/"
+  [ "${just_url_parsed_scheme}" = 'https' ]
+  [ "${just_url_parsed_authority}" = 'hub.server.com' ]
+  [ "${just_url_parsed_host}" = 'hub.server.com' ]
+  [ "${just_url_parsed_port}" = '' ]
+  [ "${just_url_parsed_rpath}" = '' ]
+  [ "${just_url_parsed_query}" = '' ]
+  [ "${just_url_parsed_fragment}" = '' ]
+
+  parse_url "https://hub.server.io:443/common/quickstart.html?highlight=T.E.S.T#what-is-t-e-s-t"
```

Comment:
Add a % escaped coded character to this, just to be complete. `%20` for space is the easiest to remember